release/release-offline.ts

```diff
@@ -0,0 +1,410 @@
+  });
+
+  if (
+    !AWS_S3_DOWNLOADS_BUCKET ||
+    !AWS_S3_STATIC_BUCKET ||
+    !AWS_S3_RELEASE_SECRET_ACCESS_KEY ||
+    !AWS_S3_RELEASE_ACCESS_KEY_ID
+  ) {
+    error("You must provide all AWS environment variables in .env-template");
+  }
+
+  process.env.AWS_ACCESS_KEY_ID = AWS_S3_RELEASE_ACCESS_KEY_ID;
+  process.env.AWS_SECRET_ACCESS_KEY = AWS_S3_RELEASE_SECRET_ACCESS_KEY;
+
+  const versionPath = edition === "ee" ? `enterprise/${version}` : version;
+
+  await $`aws s3 cp ${JAR_PATH}/metabase.jar s3://${AWS_S3_DOWNLOADS_BUCKET}/${versionPath}/metabase.jar`.pipe(
+    process.stdout,
+  );
+
+  if (isLatest) {
+    await $`aws s3 cp ${JAR_PATH}/metabase.jar s3://${AWS_S3_DOWNLOADS_BUCKET}/latest/metabase.jar`.pipe(
+      process.stdout,
+    );
+  }
+
+  if (!process.env.SKIP_CLOUDFRONT) {
+    await $`aws cloudfront create-invalidation \
+      --distribution-id ${AWS_CLOUDFRONT_DOWNLOADS_ID} \
+      --paths /${versionPath}/metabase.jar`.pipe(process.stdout);
+  }
+
+  log(`✅ Published ${version} to s3`);
+}
+
+async function docker() {
+  log(`⏳ Building docker image for ${version}`);
+
+  const { GITHUB_OWNER, GITHUB_REPO } = getGithubCredentials();
+
+  if (
+    !DOCKERHUB_RELEASE_USERNAME ||
+    !DOCKERHUB_RELEASE_TOKEN ||
+    !DOCKERHUB_OWNER
+  ) {
+    error("You must provide all docker environment variables in .env-template");
+  }
+
+  await checkJar();
+
+  await $`cp -r ${JAR_PATH}/metabase.jar ../bin/docker/`;
+  const dockerRepo = edition === "ee" ? "metabase-enterprise" : "metabase";
+  const dockerTag = `${DOCKERHUB_OWNER}/${dockerRepo}:${version}`;
+  await $`docker build --tag ${dockerTag} ../bin/docker/.`.pipe(process.stdout);
+
+  log(`⏳ Pushing docker image to dockerhub for ${version}`);
+
+  await $`docker login --username ${DOCKERHUB_RELEASE_USERNAME} -p ${DOCKERHUB_RELEASE_TOKEN}`;
+  await $`docker push ${dockerTag}`.pipe(process.stdout);
+
+  log(`✅ Published ${dockerTag} to DockerHub`);
+
+  const isLatest = await isLatestRelease({
+    github,
+    owner: GITHUB_OWNER,
+    repo: GITHUB_REPO,
+    version,
+  });
+
+  if (isLatest) {
+    const latestTag = `${DOCKERHUB_OWNER}/${dockerRepo}:latest`;
+    await $`docker tag ${dockerTag} ${latestTag}`.pipe(process.stdout);
+    await $`docker push ${latestTag}`.pipe(process.stdout);
+
+    log(`✅ Published ${latestTag} to DockerHub`);
+  }
+}
+
+async function versionInfo() {
+  log(`⏳ Building version-info.json`);
+
+  const { GITHUB_OWNER, GITHUB_REPO } = getGithubCredentials();
+
+  const newVersionInfo = await getVersionInfo({
+    github,
+    owner: GITHUB_OWNER,
+    repo: GITHUB_REPO,
+    version,
+  });
+
+  const versionInfoName =
+    edition === "ee" ? "version-info-ee.json" : "version-info.json";
+
+  fs.writeFileSync(versionInfoName, JSON.stringify(newVersionInfo, null, 2));
+
+  await $`aws s3 cp ${versionInfoName} s3://${AWS_S3_STATIC_BUCKET}/${versionInfoName}`.pipe(
+    process.stdout,
+  );
+
+  if (!process.env.SKIP_CLOUDFRONT) {
```

Comment:
`SKIP_CLOUDFRONT` is the only `process.env` that we don't destructure at the top of the file.
Also it's missing from `.env-template`.

Why would we want to `SKIP_CLOUDFRONT` anyway?

Reply:
this was convenient for testing because I don't have a parallel cloudfront environment setup. will remove